sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/ResourceExtensions.cs

```diff
@@ -33,37 +37,56 @@ internal static class ResourceExtensions
                         serviceNamespace = $"[{_serviceNamespace}]";
                         break;
                     case SemanticConventions.AttributeServiceInstance when attribute.Value is string _serviceInstance:
-                        resourceParser.RoleInstance = _serviceInstance;
-                        break;
-                    case s_AiSdkPrefixKey:
-                        SdkVersionUtils.SdkVersionPrefix = attribute.Value.ToString();
+                        azureMonitorResource.RoleInstance = _serviceInstance;
                         break;
+                    case s_AiSdkPrefixKey when attribute.Value is string _aiSdkPrefixValue:
+                        SdkVersionUtils.SdkVersionPrefix = _aiSdkPrefixValue;
+                        continue;
+                }
+
+                if (instrumentationKey != null && attribute.Key.Length <= SchemaConstants.MetricsData_Properties_MaxKeyLength && attribute.Value != null)
+                {
+                    metricsData = metricsData ?? new MetricsData(Version);
```

Comment:
nit: Move the initialization outside of this for readability?

```csharp
if (instrumentationKey != null && resource.Attributes.Count() > 0)
{
     metricsData = metricsData ?? new MetricsData(Version);
}
```

Reply:
`resource.Attributes.Count() ` will be checked everytime in the loop. We could store the count in a variable that also will impact readability.

Reply:
It also avoids the 2 check as in the if statement if first check fails it won't validate the remaining.

Reply:
> nit: Move the initialization outside of this for readability?
> 
> ```cs
> if (instrumentationKey != null && resource.Attributes.Count() > 0)
> {
>      metricsData = metricsData ?? new MetricsData(Version);
> }
> ```

This will be outside of for loop

Reply:
No strong preference, I could move it. 
Might need an additional check in the loop to check if metricsData is not null.